house-robber/pmjuu.py

```diff
@@ -0,0 +1,16 @@
+from typing import List
+
+class Solution:
+    def rob(self, nums: List[int]) -> int:
+        if len(nums) == 1:
+            return nums[0]
+        
+        # prev1: 이전 집까지의 최대 이익
+        # prev2: 전전 집까지의 최대 이익
+        prev1, prev2 = 0, 0
+        for num in nums:
+            temp = prev1
+            prev1 = max(prev2 + num, prev1) # 현재 집을 털었을 때와 안 털었을 때 중 더 큰 이익 선택
+            prev2 = temp
```

Comment:
(안 중요) 요렇게 좀 더 Pythonic 하게 코드를 작성하시면 다른 언어처럼 임시 변수를 안 쓰고 변수의 값을 스왑할 수 있을 것 같아요.

```suggestion
            prev1, prev2 = max(prev2 + num, prev1), prev1
```